src/addons/dragAndDrop/DraggableEventWrapper.js

```diff
@@ -1,51 +1,166 @@
 import cn from 'classnames'
+import compose from './compose'
 
 import BigCalendar from '../../index'
+const EventWrapper = BigCalendar.components.eventWrapper
 
-/* drag sources */
+class DraggableEventWrapper extends React.Component {
+  static propTypes = {
+    event: PropTypes.object.isRequired,
 
-let eventSource = {
-  beginDrag(props) {
-    return props.event
-  },
-}
+    connectDragSource: PropTypes.func.isRequired,
+    connectTopDragPreview: PropTypes.func.isRequired,
+    connectTopDragSource: PropTypes.func.isRequired,
+    connectBottomDragPreview: PropTypes.func.isRequired,
+    connectBottomDragSource: PropTypes.func.isRequired,
+    connectLeftDragPreview: PropTypes.func.isRequired,
+    connectLeftDragSource: PropTypes.func.isRequired,
+    connectRightDragPreview: PropTypes.func.isRequired,
+    connectRightDragSource: PropTypes.func.isRequired,
 
-function collectSource(connect, monitor) {
-  return {
-    connectDragSource: connect.dragSource(),
-    isDragging: monitor.isDragging(),
+    allDay: PropTypes.bool,
+    isRow: PropTypes.bool,
+    continuesPrior: PropTypes.bool,
+    continuesAfter: PropTypes.bool,
+    isDragging: PropTypes.bool,
+    isResizing: PropTypes.bool,
   }
-}
 
-const propTypes = {
-  connectDragSource: PropTypes.func.isRequired,
-  isDragging: PropTypes.bool.isRequired,
-  event: PropTypes.object.isRequired,
-}
+  componentDidMount() {
+    // this is needed to prevent the backend from
+    // screenshot'ing the event during a resize which
+    // would be very confusing visually
+    const emptyImage = getEmptyImage()
+    const previewOptions = { captureDraggingState: true }
+    this.props.connectTopDragPreview(emptyImage, previewOptions)
+    this.props.connectBottomDragPreview(emptyImage, previewOptions)
+    this.props.connectLeftDragPreview(emptyImage, previewOptions)
+    this.props.connectRightDragPreview(emptyImage, previewOptions)
+  }
 
-class DraggableEventWrapper extends React.Component {
   render() {
-    let { connectDragSource, isDragging, children, event } = this.props
-    let EventWrapper = BigCalendar.components.eventWrapper
+    let {
+      connectDragSource,
+      connectTopDragSource,
+      connectBottomDragSource,
+      connectLeftDragSource,
+      connectRightDragSource,
+      isDragging,
+      isResizing,
+      children,
+      event,
+      allDay,
+      isRow,
+      continuesPrior,
+      continuesAfter,
+    } = this.props
+
+    let StartAnchor = null,
+      EndAnchor = null
+
+    /*
+     * The resizability of events depends on whether they are
+     * allDay events and how they are displayed.
+     *  
+     * 1. If the event is an allDay event, we can drag it east-west
+     * to resize the days.
+     * 
+     * 2. If the event is a non-allDay event and is being displayed
+     * normally, we can drag it north-south to resize the times.
+     * 
+     * 3. If the event is a non-allDay event and is being displayed
```

Comment:
Wouldn't resizing an event, that is not all day, in month view extend the event to be more than one day long? Meaning it would extend the event to the next day at the same end time as it was previous. Am I missing something?

Reply:
@arecvlohe you're absolutely right - your suggestion behavior is perfectly reasonable, too. In fact it matches how drop (as opposed to resize) works when dropping a non-allDay event on the date header: it retains the times but changes the date. 

Great catch. I'll change the behavior and re-push when I have a moment to work on this!